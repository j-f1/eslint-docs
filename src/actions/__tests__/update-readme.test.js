jest.mock('../../spinner')

const updateReadme = require('../../actions/update-readme')

const fakeMetadata = [
  {
    name: 'foo',
    description: 'Ensure that foo is used instead of bar',
    extraDescription: 'Magical!',
    recommended: true,
  },
  {
    name: 'no-baz',
    description: 'Ensure that baz is not used',
    extraDescription: ['Unicorns', '`Rainbows`'],
  },
  {
    name: 'r',
    description: 'Require your code to be super awesome!',
    fixable: 'code',
  },
]
const fakeReadme =
  `
# eslint-plugin-test
Some description
<!-- begin rule list -->
| Name | Description | :heavy_check_mark: | :wrench: |
| ----- | ----- | ----- | ----- |
| [awesome/foo](./docs/rules/foo.md) | Ensure that foo is used instead of bar | :heavy_check_mark: | |
| [awesome/no-baz](./docs/rules/no-baz.md) | Ensure that baz is not used | | |
<!-- end rule list -->
More description
`.trim() + '\n'

const pluginName = 'awesome'
const project = { pluginName }
describe('buildBlock', () => {
  it('properly formats the rule metadata', () => {
    expect(updateReadme.buildBlock(fakeMetadata, pluginName)).toMatchSnapshot()
  })
})
describe('updateReadme', () => {
  it('replaces the block if it’s already present', () => {
    expect(updateReadme(fakeReadme, fakeMetadata, project)).toMatchSnapshot()
  })
  it('appends the block if it doesn’t exist', () => {
    expect(
      updateReadme(
        `
# eslint-plugin-test
Some description
(I’m real lazy :()
More description
        `.trim() + '\n',
        fakeMetadata,
        project
      )
    ).toMatchSnapshot()
  })
  it('is idempotent', () => {
    const firstPass = updateReadme(fakeReadme, fakeMetadata, project)
    const secondPass = updateReadme(firstPass, fakeMetadata, project)
    expect(secondPass).toEqual(firstPass)
  })
  it('is case-insensitive', () => {
    expect(
      updateReadme(fakeReadme.toUpperCase(), fakeMetadata, project)
    ).toMatchSnapshot()
  })
})
