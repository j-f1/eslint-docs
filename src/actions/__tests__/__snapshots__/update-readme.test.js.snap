// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildBlock properly formats the rule metadata 1`] = `
"<!-- begin rule list -->
**Key**: :heavy_check_mark: = recommended, :wrench: = fixable
| Name | Description | :heavy_check_mark: | :wrench: |
| ---- | ----------- | ------------------ | -------- |
| [\`awesome/foo\`](./docs/rules/foo.md) | Ensure that foo is used instead of bar (Magical!) | :heavy_check_mark: |  |
| [\`awesome/no-baz\`](./docs/rules/no-baz.md) | Ensure that baz is not used (Unicorns, \`Rainbows\`) |  |  |
| [\`awesome/r\`](./docs/rules/r.md) | Require your code to be super awesome! |  | :wrench: |
<!-- end rule list -->"
`;

exports[`updateReadme appends the block if it doesn’t exist 1`] = `
"# eslint-plugin-test
Some description
(I’m real lazy :()
More description


<!-- begin rule list -->
**Key**: :heavy_check_mark: = recommended, :wrench: = fixable
| Name | Description | :heavy_check_mark: | :wrench: |
| ---- | ----------- | ------------------ | -------- |
| [\`awesome/foo\`](./docs/rules/foo.md) | Ensure that foo is used instead of bar (Magical!) | :heavy_check_mark: |  |
| [\`awesome/no-baz\`](./docs/rules/no-baz.md) | Ensure that baz is not used (Unicorns, \`Rainbows\`) |  |  |
| [\`awesome/r\`](./docs/rules/r.md) | Require your code to be super awesome! |  | :wrench: |
<!-- end rule list -->
"
`;

exports[`updateReadme is case-insensitive 1`] = `
"# ESLINT-PLUGIN-TEST
SOME DESCRIPTION
<!-- begin rule list -->
**Key**: :heavy_check_mark: = recommended, :wrench: = fixable
| Name | Description | :heavy_check_mark: | :wrench: |
| ---- | ----------- | ------------------ | -------- |
| [\`awesome/foo\`](./docs/rules/foo.md) | Ensure that foo is used instead of bar (Magical!) | :heavy_check_mark: |  |
| [\`awesome/no-baz\`](./docs/rules/no-baz.md) | Ensure that baz is not used (Unicorns, \`Rainbows\`) |  |  |
| [\`awesome/r\`](./docs/rules/r.md) | Require your code to be super awesome! |  | :wrench: |
<!-- end rule list -->
MORE DESCRIPTION
"
`;

exports[`updateReadme replaces the block if it’s already present 1`] = `
"# eslint-plugin-test
Some description
<!-- begin rule list -->
**Key**: :heavy_check_mark: = recommended, :wrench: = fixable
| Name | Description | :heavy_check_mark: | :wrench: |
| ---- | ----------- | ------------------ | -------- |
| [\`awesome/foo\`](./docs/rules/foo.md) | Ensure that foo is used instead of bar (Magical!) | :heavy_check_mark: |  |
| [\`awesome/no-baz\`](./docs/rules/no-baz.md) | Ensure that baz is not used (Unicorns, \`Rainbows\`) |  |  |
| [\`awesome/r\`](./docs/rules/r.md) | Require your code to be super awesome! |  | :wrench: |
<!-- end rule list -->
More description
"
`;
